lib/ansiblelint/rules/MetaTagValidRule.py

```diff
@@ -57,3 +58,23 @@ def matchplay(self, file, data):
                                 "{}, invalid: '{}'".format(msg, tag)))
 
         return results
+
+
+META_TAG_VALID = '''
+galaxy_info:
+    galaxy_tags: ['database', 'my s q l', 'MYTAG']
+    categories: 'my_category_not_in_a_list'
+'''
+
+if "pytest" in sys.modules:
+
+    import pytest
+
+    @pytest.mark.parametrize('rule_runner', (MetaTagValidRule, ), indirect=['rule_runner'])
+    def test_valid_tag_rule(rule_runner):
```

Comment:
I don't like that this approach adds an extra indent. Maybe we could just have an extra file with RuleTest instead of Rule in the name?

Reply:
The whole point of this move is to encapsulate testing of rules in a single file. This would easy contributions and also allow us to include tests as examples/documentation. We will be able to expose them in docs and command like.

Take a look at eslint docs, random rule documentation: https://eslint.org/docs/rules/no-import-assign -- it easy to read with good and bad usage examples, exactly what we do to test the rules.